examples/basic_dispatcher.py

```diff
@@ -1,33 +1,60 @@
 # SPDX-License-Identifier: MIT
 
-# This example uses disnake
+# Basic functionality example using disnake.
+
+import os
 
 import disnake
 from disnake.ext import commands
 
 from dispatcher import Dispatcher
 
-bot = commands.Bot(command_prefix=commands.when_mentioned, intents=None)
-bot.dispatcher = Dispatcher(bot, library_name=disnake.__name__)
 
+# We don't need prefix commands, so we are using InteractionBot here.
+class MyBot(commands.InteractionBot):
+    # Dispatcher uses generics so that you can use any bot
+    # class from any library: you just need to type-hint it.
+    dispatcher: Dispatcher["MyBot"]
+
+    def __init__(self) -> None:
+        super().__init__(intents=disnake.Intents.default())
+        # One would usually assign dispatcher to the bot directly.
+        self.dispatcher = Dispatcher(self)
+
+    async def on_ready(self) -> None:
+        print(f"Ready! Logged in as {self.user}")
+
+        # Send message to any available channel.
+        for guild in self.guilds:
+            for channel in guild.text_channels:
+                await channel.send("Hello!")
+                break
+            break
 
-@bot.event
-async def on_ready():
-    print(bot.user)
-    bot.dispatcher.dispatch("my_event")
+    async def on_message(self, message: disnake.Message) -> None:
+        # The message is sent by us.
+        if message.author.id == self.user.id:
+            # Dispatch the 'on_self_message' event.
+            self.dispatcher.dispatch("on_self_message", message)
 
-    # passing args and kwargs
-    bot.dispatcher.dispatch("my_second_event", 10, name="Snipy")
+    # Event listeners can be defined on bot class itself..
+    async def on_self_message(self, message: disnake.Message):
+        print(f"Self-sent message at {message.created_at}")
 
 
-@bot.listen("on_my_event")
-async def foo():
-    print("on_my_event was called")
+bot = MyBot()
 
 
-@bot.listen("on_my_second_event")
-async def bar(n, *, name):
-    print("on_my_second_event was called", n, name)
+# ..or be registered as a listener, either via @commands.Bot.listen or
+# @commands.Cog.listener() decorators..
```

Comment:
we could also metion that it works using `@commands.Bot.event` (so also for other subclasses) too

Reply:
It is shown above, in the definition of MyBot, see on_self_message method